### PR TITLE
Update backtest data

### LINKED ISSUE #32: Update backtest with live data
- Use live squeeth data for IV on the backtest from launch onwards
- Clearly delineate the difference on the graph (maybe the line becomes a different color?)

### PR DESCRIPTION
**Pull request by block902**
# Task: Update backtest data for squeeth PNL graph.

## Description
- Add hook to get NormFactor History from timestamps array
- Add hook to get squeeth PNL data with live VOL
- Change funding graph label from `Annual` to `Annualized`

Fixes #32 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Tested locally

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

### CODE REVIEW

**Review by alexisgauba — CHANGES_REQUESTED** [review]
Hmm I'm not able to see the squeeth PnL line, only the ETH one 🤔 

Screen Shot 2022-02-25 at 10 37 56 AM

**Comment by block902** [comment]
> https://continuouscall-git-fe-update-backtest-data-opynfinance.vercel.app

Currently it takes more time to get live data because it is looping all ETH prices and fetching Norm Factor data for the timestamp of every ETH price.

**Comment by block902** [comment]
> Hmm I'm not able to see the squeeth PnL line, only the ETH one 🤔
> 
> Screen Shot 2022-02-25 at 10 37 56 AM
Currently it takes more time to get live data because it is looping all ETH prices and fetching Norm Factor data for the timestamp of every ETH price.

**Review by yichiehliu — COMMENTED** [review]
image

It seemed that it was taking a bit long to load, the graph loaded for 5mins to show up for me, thinking that it's not perfect in terms of user experience
Do we want to maybe store the data in firebase and update the new data to the firebase like getVolMap did instead of iterating all the data every time users enter trade page? Would love to have y'all thoughts on it

**Comment by block902** [comment]
> as taking a bit long to load, the graph loaded for 5mins to show up for me, thinking that it's not perfect in terms of user experience
> Do we want to maybe store the data in firebase and update the new data to the firebase like getVolMap did instead of iterating all the data every time users enter trade page? Would love to have y'all thoughts on it

We are getting Norm Factors for ETH prices in subgraph with with looping all timestamps and getting the item in subgraph between modified timestamp and current timestamp.
i.e. There are 366 items in ETH prices array for current year and we need to get Norm Factors from subgraph of every item in the array. So we need to call the query to get Norm Factor data from subgraph 366 times. 
That's why it takes long to load all data. Please let me know if you have any solution for this. Thanks.

**Comment by alexisgauba** [comment]
> https://continuouscall-git-fe-update-backtest-data-opynfinance.vercel.app

Yeah agree that its taking too long which won't work. Thoughts on @yichiehliu's firebase suggestion? 

also cc @KMKoushik @EstherSeyi @akwritescode to get ya'll thoughts as well

**Review by alexisgauba — CHANGES_REQUESTED** [review]
Colors looks good! Could we change the box color here to be yellow when live is true? 

Screen Shot 2022-02-28 at 9 28 59 AM

**Comment by akwritescode** [comment]
agree it's too slow. like @block902 mentioned in slack, we probably should load from firebase.

**Review by yichiehliu — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/26922518/156042475-b89049b6-3093-4253-aec0-d3eaba7d6c4d.mov
When I updated the historical day input, the graph sometimes not responded

File: packages/frontend/src/components/Charts/LongChart.tsx [hunk 1]
```diff
@@ -296,7 +305,11 @@ export function LongChart() {
               </div>
               <div className={classes.legendContainer}>
                 <div style={{ width: '20px', height: '20px', backgroundColor: '#00E396' }}></div>
-                <div>Squeeth PNL</div>
+                <div>Squeeth Deribit PNL</div>
+              </div>
+              <div className={classes.legendContainer}>
+                <div style={{ width: '20px', height: '20px', backgroundColor: '#FEB01B' }}></div>
```
Comment (adam5909099) [inline packages/frontend/src/components/Charts/LongChart.tsx:311]: It's better to create a reusable component for 20*20 box which receives background color

File: packages/frontend/src/hooks/useETHPriceCharts.tsx [hunk 2]
```diff
@@ -168,30 +168,35 @@ export function useETHPriceCharts(initDays = 365, initVolMultiplier = 1.2, initC
...
         return { time, value: longPNL }
       })
     )
-  }, [squeethPNLSeries.data])
+  }, [squeethPNLSeries])
+
+  const squeethIsLive = useMemo(() => {
+    return (
+      squeethPNLSeries &&
+      squeethPNLSeries.map(({ isLive }) => {
```
Comment (adam5909099) [inline packages/frontend/src/hooks/useETHPriceCharts.tsx:185]: I'm not sure about team's preference but we can use optional chaining here.

Comment (block902) [reply packages/frontend/src/hooks/useETHPriceCharts.tsx:185]: Could you tell me in details about optional chaining?

**Review by yichiehliu — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/26922518/156415044-99e96f4f-e93c-4720-8eba-8101cbb831df.mov

It seemed that when I changed the input of historical day, the graph sometimes wouldn't load for me tho

**Comment by block902** [comment]
> 2022-03-02.9.24.54.mov 
> It seemed that when I changed the input of historical day, the ETH PNL sometimes wouldn't load for me tho

Fixed this issue to load ETH and Squeeth PNL at the same time.

**Review by yichiehliu — APPROVED** [review]

**Pull request merged**